clientd3d/d3drender.c

```diff
@@ -7064,16 +7059,35 @@ void D3DRenderObjectsDraw(d3d_render_pool_new *pPool, room_type *room,
 				pChunk->flags |= D3DRENDER_NOAMBIENT;
 		}
 
+		bool translucent = false;
 		if (GetDrawingEffect(pRNode->obj.flags) == OF_TRANSLUCENT25)
+		{
 			bgra.a = D3DRENDER_TRANS25;
+			translucent = true;
+		}
 		if (GetDrawingEffect(pRNode->obj.flags) == OF_TRANSLUCENT50)
+		{
 			bgra.a = D3DRENDER_TRANS50;
+			translucent = true;
+		}
 		if (GetDrawingEffect(pRNode->obj.flags) == OF_TRANSLUCENT75)
+		{
 			bgra.a = D3DRENDER_TRANS75;
-		if (GetDrawingEffect(pRNode->obj.flags) == OF_DITHERTRANS)
+			translucent = true;
+		}
+		if (GetDrawingEffect(pRNode->obj.flags) == OF_DITHERTRANS)	
+		{
 			bgra.a = D3DRENDER_TRANS50;
+			translucent = true;
+		}
 		if (GetDrawingEffect(pRNode->obj.flags) == OF_DITHERINVIS)
+		{
 			bgra.a = D3DRENDER_TRANS50;
+			translucent = true;
+		}
+
+		if(translucent)

```

Comment:
I think you can check bgra.a here directly rather than introducing the "translucent" variable.